apps/rush-lib/src/cli/utilities/PublishUtilities.ts

```diff
@@ -205,8 +206,13 @@ export default class PublishUtilities {
   ): string {
     const currentDependencyVersion: string = dependencies[dependencyName];
     let newDependencyVersion: string;
-
     if (PublishUtilities.isRangeDependency(currentDependencyVersion)) {
+      // If version is pre-release and current dependency does not have a prerelease tag,
+      // replace dependency to match the exact pre-release version
+      const HAS_PRERELEASE_VERSION: RegExp = /-.+\s/;
+      if (!HAS_PRERELEASE_VERSION.test(currentDependencyVersion) && semver.prerelease(newProjectVersion)) {
+        return newProjectVersion;
+      }
```

Comment:
Prerelease does not need to be exact match. as long as it is part of the range.

I do not want hotfix logic to mess up existing version bump logic.